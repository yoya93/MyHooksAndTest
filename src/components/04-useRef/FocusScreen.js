import React, { useRef } from "react";
import "../02-useEffect/simpleForm.css";

export const FocusScreen = () => {
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <>
      <h1>Focus Screen</h1>
      <hr />

      <input ref={inputRef} className="form-control" placeholder="Su nombre" />

      <button className="btn btn-primary mt-5" onClick={handleClick}>
        Focus
      </button>
    </>
  );
};
