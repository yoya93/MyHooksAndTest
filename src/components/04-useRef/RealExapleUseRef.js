import React, { useState } from "react";
import "../02-useEffect/simpleForm.css";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";

export const RealExapleUseRef = () => {
  const [showComponet, setShowComponet] = useState(false);

  return (
    <>
      <h1>Focus Screen</h1>
      <hr />

      {showComponet && <MultipleCustomHooks />}

      <button
        className="btn btn-primary mt-5 inline-block"
        onClick={() => {
          setShowComponet(!showComponet);
        }}
      >
        Show
      </button>
    </>
  );
};
