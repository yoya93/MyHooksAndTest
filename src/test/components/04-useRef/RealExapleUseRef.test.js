import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { RealExapleUseRef } from "../../../components/04-useRef/RealExapleUseRef";

describe("Pruebas en el <RealExapleUseRef>", () => {
  const wrapper = shallow(<RealExapleUseRef />);
  test("debe hacer match el componente RealExapleUseRef ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el componente <MultipleCustomHooks />", () => {
    const wrapper = shallow(<RealExapleUseRef />);

    wrapper.find("button").simulate("click");

    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
