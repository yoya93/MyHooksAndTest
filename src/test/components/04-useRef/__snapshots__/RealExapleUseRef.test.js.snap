// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <RealExapleUseRef> debe hacer match el componente RealExapleUseRef  1`] = `
<Fragment>
  <h1>
    Focus Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5 inline-block"
    onClick={[Function]}
  >
    Show
  </button>
</Fragment>
`;
